kube-runtime/src/controller/mod.rs

```diff
@@ -417,6 +419,29 @@ where
     }
 }
 
+/// Accumulates all options that can be used on a [`Controller`] invocation.
+#[derive(Clone, Debug, Default)]
+pub struct Config {
+    debounce: Duration,
+}
+
+impl Config {
+    /// The debounce duration used to deduplicate reconciliation requests.
+    ///
+    /// When set to a non-zero duration, debouncing is enabled in the [`Scheduler`] resulting
+    /// in __trailing edge debouncing__ of reqonciler requests.
+    /// This option can help to reduce the amount of unnecessary reconciler calls
+    /// when using multiple controller relations, or during rapid phase transitions.
+    ///
+    /// ## Warning
+    /// This option delays (and keeps delaying) reconcile requests for objects while
+    /// the object is updated. It can **permanently hide** updates from your reconciler
+    /// if set too high on objects that are updated frequently (like nodes).
+    pub fn debounce(&mut self, debounce: Duration) {
+        self.debounce = debounce;
+    }
```

Comment:
```suggestion
    pub fn debounce(&mut self, debounce: Duration) -> Self{
        self.debounce = debounce;
        self
    }
```
this needs to be a builder